File: app/src/main/java/ykk/cb/com/cbwms/produce/adapter/Prod_WorkBySaoMa_SearchFragment1Adapter.java
package ykk.cb.com.cbwms.produce.adapter;

import android.app.Activity;
import android.text.Html;
import android.view.View;
import android.widget.TextView;

import java.text.DecimalFormat;
import java.util.List;

import ykk.cb.com.cbwms.R;
import ykk.cb.com.cbwms.model.WorkRecordSaoMaEntry1;
import ykk.cb.com.cbwms.model.WorkRecordSaoMaEntry1;
import ykk.cb.com.cbwms.util.basehelper.BaseArrayRecyclerAdapter;

public class Prod_WorkBySaoMa_SearchFragment1Adapter extends BaseArrayRecyclerAdapter<WorkRecordSaoMaEntry1> {
    private DecimalFormat df = new DecimalFormat("#.######");
    private Activity context;
    private MyCallBack callBack;

    public Prod_WorkBySaoMa_SearchFragment1Adapter(Activity context, List<WorkRecordSaoMaEntry1> datas) {
        super(datas);
        this.context = context;
    }

    @Override
    public int bindView(int viewtype) {
        return R.layout.prod_work_saoma_search_fragment1_item;
    }

    @Override
    public void onBindHoder(RecyclerHolder holder, final WorkRecordSaoMaEntry1 entity, final int pos) {
        // 初始化id
        TextView tv_row = holder.obtainView(R.id.tv_row);
        TextView tv_mtlPriceType = holder.obtainView(R.id.tv_mtlPriceType);
        TextView tv_process = holder.obtainView(R.id.tv_process);
        TextView tv_num = holder.obtainView(R.id.tv_num);
        TextView tv_price = holder.obtainView(R.id.tv_price);
        TextView tv_money = holder.obtainView(R.id.tv_money);

        // 赋值
        tv_row.setText(String.valueOf(pos + 1));
        tv_mtlPriceType.setText(entity.getMtlPriceTypeName());
        tv_process.setText(entity.getProcedureName());
        if(entity.getStrLocationQty().indexOf(":") > -1) {
            tv_num.setText(Html.fromHtml("<small>"+entity.getStrLocationQty()+"</small>"));
        } else {
            tv_num.setText(Html.fromHtml(entity.getStrLocationQty()));
        }
        if(entity.getStrLocationPrice().indexOf(":") > -1) {
            tv_price.setText(Html.fromHtml("<small>"+entity.getStrLocationPrice()+"</small>"));
        } else {
            tv_price.setText(Html.fromHtml(entity.getStrLocationPrice()));
        }
        tv_money.setText(df.format(entity.getSumMoney()));

    }

    public void setCallBack(MyCallBack callBack) {
        this.callBack = callBack;
    }

    public interface MyCallBack {
//        void onClick_num(WorkRecordSaoMaEntry1 entity, int position);
    }


}

File: app/src/main/java/ykk/cb/com/cbwms/produce/adapter/Prod_WorkBySaoMa_SearchFragment1Adapter.kt
package ykk.cb.com.cbwms.produce.adapter

import android.app.Activity
import android.text.Html
import android.widget.TextView
import ykk.cb.com.cbwms.R
import ykk.cb.com.cbwms.model.WorkRecordSaoMaEntry1
import ykk.cb.com.cbwms.util.basehelper.BaseArrayRecyclerAdapter
import ykk.cb.com.cbwms.util.basehelper.BaseRecyclerAdapter
import java.text.DecimalFormat

class Prod_WorkBySaoMa_SearchFragment1Adapter(private val context: Activity, datas: List<WorkRecordSaoMaEntry1>) : BaseArrayRecyclerAdapter<WorkRecordSaoMaEntry1>(datas) {
    private val df = DecimalFormat("#.######")
    private var callBack: MyCallBack? = null

    override fun bindView(viewtype: Int): Int {
        return R.layout.prod_work_saoma_search_fragment1_item
    }

    override fun onBindHoder(holder: BaseRecyclerAdapter.RecyclerHolder, entity: WorkRecordSaoMaEntry1, pos: Int) {
        // 初始化id
        val tv_row = holder.obtainView(R.id.tv_row) as TextView
        val tv_mtlPriceType = holder.obtainView(R.id.tv_mtlPriceType) as TextView
        val tv_process = holder.obtainView(R.id.tv_process) as TextView
        val tv_num = holder.obtainView(R.id.tv_num) as TextView
        val tv_price = holder.obtainView(R.id.tv_price) as TextView
        val tv_money = holder.obtainView(R.id.tv_money) as TextView

        // 赋值
        tv_row.text = (pos + 1).toString()
        tv_mtlPriceType.text = entity.mtlPriceTypeName
        tv_process.text = entity.procedureName
        if (entity.strLocationQty.indexOf(":") > -1) {
            tv_num.text = Html.fromHtml("<small>" + entity.strLocationQty + "</small>")
        } else {
            tv_num.text = Html.fromHtml(entity.strLocationQty)
        }
        if (entity.strLocationPrice.indexOf(":") > -1) {
            tv_price.text = Html.fromHtml("<small>" + entity.strLocationPrice + "</small>")
        } else {
            tv_price.text = Html.fromHtml(entity.strLocationPrice)
        }
        tv_money.text = df.format(entity.sumMoney)
    }

    fun setCallBack(callBack: MyCallBack) {
        this.callBack = callBack
    }

    interface MyCallBack//        void onClick_num(WorkRecordSaoMaEntry1 entity, int position);


}
